translator/translate/otel/processor/metricstransformprocesssor/translator.go

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricstransformprocessor"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+type translator struct {
+	name    string
+	factory processor.Factory
+}
+
+var _ common.Translator[component.Config] = (*translator)(nil)
+
+func NewTranslatorWithName(name string) common.Translator[component.Config] {
+	return &translator{name, metricstransformprocessor.NewFactory()}
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	cfg := t.factory.CreateDefaultConfig().(*metricstransformprocessor.Config)
+	c := confmap.NewFromStringMap(map[string]interface{}{
```

Comment:
This logic should not apply to ECS and more specifically, even for EKS, it should only apply when enhanced container insights is enabled.

Reply:
I added the metricstransformprocessor only to the containerinsights pipeline, is that not enough ?

Reply:
Okay so I moved this logic to the containerinsights pipeline. if the enhanced container insights flag is present, then the metricstransform processor will be appended to the pipeline